src/plugins/visualizations/server/saved_objects/visualization_migrations.ts

```diff
@@ -981,6 +982,78 @@ const removeDefaultIndexPatternAndTimeFieldFromTSVBModel: SavedObjectMigrationFn
   return doc;
 };
 
+const addEmptyValueRuleForSavedObjectsWithLessAndGreaterThenZeroRules: SavedObjectMigrationFn<
+  any,
+  any
+> = (doc) => {
+  const visStateJSON = get(doc, 'attributes.visState');
+  let visState;
+
+  if (visStateJSON) {
+    try {
+      visState = JSON.parse(visStateJSON);
+    } catch (e) {
+      // Let it go, the data is invalid and we'll leave it as is
+    }
+
+    if (visState && visState.type === 'metrics') {
+      const params: any = get(visState, 'params') || {};
+      const {
+        bar_color_rules: barColorRules = [],
+        background_color_rules: backgroundColorRules = [],
+        gauge_color_rules: gaugeColorRules = [],
+      } = params;
+
+      const getRuleWithComparingToZero = (rules: any[] = []) => {
+        const compareWithEqualMethods = ['lte', 'gte'];
+        return last(
+          rules.filter(
+            (rule) => compareWithEqualMethods.includes(rule.operator) && rule.value === 0
+          )
+        );
+      };
+
+      const convertRuleToEmpty = (rule: any = {}) => ({
+        ...rule,
+        id: uuid.v4(),
+        operator: 'empty',
+        value: null,
+      });
+
+      const addEmptyRuleToListIfNecessary = (rules: any[]) => {
+        const rule = getRuleWithComparingToZero(rules);
+
+        if (rule) {
+          return [...rules, convertRuleToEmpty(rule)];
+        }
+
+        return rules;
+      };
+
+      const colorRules = {
+        bar_color_rules: addEmptyRuleToListIfNecessary(barColorRules),
```

Comment:
nit: `barColorRules` constant looks a little bit extra for me 

I think the code below does not serve any purpose
```
const {
        bar_color_rules: barColorRules = [],
        background_color_rules: backgroundColorRules = [],
        gauge_color_rules: gaugeColorRules = [],
      } = params;
```

what if we simply pass `params['bar_color_rules']` into `addEmptyRuleToListIfNecessary`